pkg/cmd/roachtest/operations/changefeeds/create.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package changefeeds
+
+import (
+	"context"
+	gosql "database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeedbase"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/operation"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/operations/helpers"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestflags"
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+)
+
+// runCreateChangefeeds creates a new changefeed job if the number of current changefeeds is below the allowed maximum.
+func runCreateChangefeeds(
+	ctx context.Context, o operation.Operation, c cluster.Cluster,
+) registry.OperationCleanup {
+	// Establish a connection to the cluster.
+	conn := c.Conn(ctx, o.L(), 1, option.VirtualClusterName(roachtestflags.VirtualCluster))
+	defer func() {
+		_ = conn.Close()
+	}()
+
+	// Ensure rangefeed is enabled in the cluster, enable it if not.
+	err := enableRangeFeedIfNotDoneAlready(ctx, conn)
+	if err != nil {
+		o.Fatal(err)
+	}
+
+	// Fetch all changefeed jobs that are running or paused.
+	allCFJobs, err := fetchAllChangefeedJobsWithStatuses(ctx, conn, jobs.StatusRunning, jobs.StatusPaused)
+	if err != nil {
+		o.Fatal(err)
+	}
+
+	// Get the max allowed changefeeds from environment variables.
+	mcf, err := helpers.EnvOrDefaultInt(maxChangefeeds, defaultEnvValuesInt[maxChangefeeds])
+	if err != nil {
+		o.Fatal(err)
+	}
+
+	// If the number of existing changefeeds is below the max limit, create a new one.
+	if len(allCFJobs) < mcf {
+		// Select a random database and table to create the changefeed for.
+		dbName := helpers.PickRandomDB(ctx, o, conn, helpers.SystemDBs)
```

Comment:
idk if all those dbs can have changefeeds. can you stick to dbs with workloads running on them?

Reply:
This is configurable as a flag to the operation. We set the flag when we run the operations.